### PR TITLE
Add Java and Node versions supported

### PR DESCRIPTION
**Pull request by mraible**
Fixes https://github.com/jhipster/generator-jhipster/issues/23933.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,15 @@ Please read our [guidelines](/CONTRIBUTING.md#submitting-an-issue) before submit
...
 We are honoured by any contributions you may have small or large. Please refer to our [contribution guidelines and instructions document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) for any information about contributing to the project.
 
+## Java and Node versions
+
+The following Java and Node combinations are tested and verified by GitHub Actions:
+
+| Java | Node | Status |
+|------|------|--- ----|
+| 21   | 20   | ✅    |
+| 17   | 18   | ✅    |
```
Comment (mraible) [inline README.md:24]: @mshima It'd be awesome if the status icon could be dynamic based on the matrix results.

Comment (mshima) [reply README.md]: I don’t think it’s possibe.

Comment (mraible) [reply README.md]: Is this table accurate or does it need to be updated?

Comment (mshima) [reply README.md]: > Is this table accurate or does it need to be updated?

I don't know how are we going to add the node/java matrix yet.
From the table I would think that we are test node20/java21 together and node18/java17 together.
Node 20 should be ok to generate applications, but test is broken.

Comment (mshima) [reply README.md]: Maybe we should create the matrix workflows at daily builds.
We can add a workflow for each combination, badges will be available.

Comment (mraible) [reply README.md]: Sounds good to me!

**Review by DanielFran — APPROVED** [review]

**Pull request merged**